lib/hooks/useDebounceTerm.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useRef, useState } from "react";
+
+const useDebounceTerm = (value: string, delay = 500) => {
+  const [debouncedValue, setDebouncedValue] = useState("");
+  const timeoutRef = useRef();
+
+  useEffect(() => {
+    timeoutRef.current = setTimeout(() => setDebouncedValue(value), delay) as unknown as typeof timeoutRef.current;
```

Comment:
Refactor (blocking): use the native type.

```suggestion
  const timeoutRef = useRef<NodeJS.Timeout>();

  useEffect(() => {
    timeoutRef.current = setTimeout(() => setDebouncedValue(value), delay);
```